autoarray/structures/grids.py

```diff
@@ -1317,93 +1220,142 @@ def interpolated_values_from_values(self, values):
+        a = 0
 
-        coordinate_1d_index = 0
+        for coords in coordinates:
+            a = a + len(coords)
+            upper_indexes.append(a)
 
-        new_coordinates = []
+        coordinates_arr = np.concatenate([np.array(i) for i in coordinates])
 
-        for coordinate_set_index in range(len(self)):
-            new_coordinate_set_list = []
-            for coordinates_index in range(len(self[coordinate_set_index])):
-                new_coordinate_set_list.append(
-                    tuple(coordinates_1d[coordinate_1d_index, :])
-                )
-                coordinate_1d_index += 1
-            new_coordinates.append(new_coordinate_set_list)
+        obj = coordinates_arr.view(cls)
+        obj.upper_indexes = upper_indexes
+        obj.lower_indexes = [0] + upper_indexes[:-1]
+        obj.mask = mask
 
-        return Coordinates(coordinates=new_coordinates, mask=self.mask)
+        return obj
 
-    def from_1d_values(self, values_1d):
+    def __array_finalize__(self, obj):
 
-        values_1d_index = 0
+        if hasattr(obj, "lower_indexes"):
+            self.lower_indexes = obj.lower_indexes
 
-        new_values_list = []
+        if hasattr(obj, "upper_indexes"):
+            self.upper_indexes = obj.upper_indexes
 
-        for coordinate_set_index in range(len(self)):
-            new_values_set_list = []
-            for coordinates_index in range(len(self[coordinate_set_index])):
-                new_values_set_list.append(values_1d[values_1d_index])
-                values_1d_index += 1
-            new_values_list.append(new_values_set_list)
+        if hasattr(obj, "mask"):
+            self.mask = obj.mask
 
-        return new_values_list
+    @classmethod
+    def from_yx_1d(cls, y, x):
+        """Create *Coordinates* from a list of y and x values.
+
+        This function omits coordinate grouping."""
+        return Coordinates(coordinates=np.stack((y, x), axis=-1))
+
+    @classmethod
+    def from_pixels_and_mask(cls, pixels, mask):
+        """Create *Coordinates* from a list of coordinates in pixel units and a mask which allows these coordinates to
+        be converted to scaled units."""
+        coordinates = []
+        for coordinate_set in pixels:
+            coordinates.append(
+                [
+                    mask.geometry.scaled_coordinates_from_pixel_coordinates(
+                        pixel_coordinates=coordinates
+                    )
+                    for coordinates in coordinate_set
+                ]
+            )
+        return cls(coordinates=coordinates, mask=mask)
 
     @property
-    def scaled(self):
+    def in_1d(self):
         return self
 
     @property
-    def pixels(self):
+    def in_list(self):
+        """Return the coordinates on a structured list which groups coordinates with a common origin."""
+        return [
+            list(map(tuple, self[i:j, :]))
+            for i, j in zip(self.lower_indexes, self.upper_indexes)
```

Comment:
There's a lot of repetition between these methods and those in values. Might make sense to have a common mixin to implement these methods?

Reply:
Agreed, I am expecting this to all end up looking like that.